docs/logging.md

```diff
@@ -0,0 +1,204 @@
+# Contributor Guidelines - Logging
+
+## Framework
+
+We use structured logging, and specifically we use [Zap](https://github.com/uber-go/zap)
+as our logging framework.
+
+Why not something else?
+
+- [Zerolog](https://github.com/rs/zerolog) is more performant and better
+  looking, but its syntax is more error-prone (no logging in case you forget
+  to call `Msg("...")` at the end)
+- Go's [Slog](https://pkg.go.dev/log/slog) is new, slower (according to Zap
+  benchmarks), and doesn't provide useful benefits
+
+## Default Level
+
+The default minimum logging level is [Info](https://pkg.go.dev/go.uber.org/zap/zapcore#InfoLevel).
+To change this, set the following AutoKitteh config value to the lower-case or
+all-caps name of the level you want. For example:
+
+- CLI: `ak config set logger.level debug`
+
+- Config YAML file:
+
+  ```yaml
+  logger:
+    level: debug
+  ```
+
+- Environment variable: `AK_LOGGER__LEVEL=DEBUG`
+
+## Logger Naming
+
+- Local variables: `l` is preferable when its meaning is clear, otherwise
```

Comment:
I don't like `l`, it is easily confused with `I` or `1`. See [Lower Case l Looks a Lot Like the Digit 1](https://cs.fit.edu/~kgallagher/Schtick/How%20To%20Write%20Unmaintainable%20Code.html)

Reply:
I traditionally use `z` for Zap, but I can see why some people are irked by that. I think `logger` is too long. `l` makes sense since it always written as `l.Info` or some such, so not so easily confused with 1 or I.